Typed Template Haskell/article.md

```diff
@@ -295,6 +302,41 @@ Annotating `mempty'` may resolve it in this case:
 
 An [open ticket](https://gitlab.haskell.org/ghc/ghc/-/issues/10271) exists describing the issue, but if you run into some strange errors, it's a good idea to keep it in mind.
 
+## Compatibility with GHC 9
+
+In GHC 9, the types used by typed Template Haskell were changed according to the [Make Q (TExp a) into a newtype](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0195-code-texp.rst) proposal.
+
+This proposal changed the typed expression quasi-quoter so it now returns `Quote m => Code m a` instead of the old `Quote m => m (TExp a)` from [`template-haskell-2.17.0.0`](https://hackage.haskell.org/package/template-haskell-2.17.0.0/docs/Language-Haskell-TH.html#t:Code).
```

Comment:
I don’t understand the link to th-2.17. I think, `Code` was implemented exactly at the same time as the overloaded bracket.

In other words, what I am saying is that the proposal you are referring indeed changed the type from `Quote m => m (TExp a)` to `Quote m => Code m a`, but also, a the exact same time another proposal changed the type from `Q (TExp a)` to `Quote m => m (TExp a)`, so the reference to 2.17 does not seem relevant, as both changes happened in there.

To sum up, IMHO, actionable points here are:

1. Maybe mention that it used to be `Q (TExp a)` prior to GHC 9.
2. Remove the link to template-haskell? I mean, I genuinely don’t understand what it means, maybe I am missing something.

Reply:
I've removed the link and used `Q (TExp a)` there.

Reply:
Ok, but now the connection between the old time and the new type is harder to grasp... at least without any context. Maybe it would be worth it to somehow point out that `m` is what used to be `Q` before the changes?

Reply:
I elaborated a bit more. I think it should be clearer now.